cpp/arcticdb/storage/rocksdb/rocksdb_storage.cpp

```diff
@@ -125,10 +126,18 @@ void RocksDBStorage::do_read(Composite<VariantKey>&& ks, const ReadVisitor& visi
             std::string k_str = to_serialized_key(k);
             std::string value;
             auto s = db_->Get(::rocksdb::ReadOptions(), handle, ::rocksdb::Slice(k_str), &value);
-            util::check(s.ok(), DEFAULT_ROCKSDB_NOT_OK_ERROR + s.ToString());
-            visitor(k, Segment::from_bytes(reinterpret_cast<uint8_t*>(value.data()), value.size()));
+            storage::check<ErrorCode::E_UNEXPECTED_ROCKSDB_ERROR>(s.IsNotFound() || s.ok(), DEFAULT_ROCKSDB_NOT_OK_ERROR + s.ToString());
+            if(s.IsNotFound()) {
+                log::storage().debug("Failed to read segment for key {}", variant_key_view(k));
```

Comment:
Can you please mirror the [S3 behaviour](https://github.com/man-group/ArcticDB/blob/bb2278d845abef2b7e7101dc968f0974cf7bdde1/cpp/arcticdb/storage/s3/detail-inl.hpp#L182) which is to warn by default, except if a flag is set in the `ReadKeyOpts`. This is relevant for reading the incompletes linked list, where we explicitly want this to go to debug and not warn since this is expected at the end of the linked list - see #1130 for more details.

Having warn by default will help to debug any errors where the `KeyNotFoundException` is swallowed by the caller, since by default all log [levels are set to info](https://docs.arcticdb.io/latest/runtime_config/#logging-configuration) ("By default, all streams are set to the INFO level").

Reply:
Fixed in [9fed291](https://github.com/man-group/ArcticDB/pull/1360/commits/9fed291316113f706a553493480ea20ed0937916)